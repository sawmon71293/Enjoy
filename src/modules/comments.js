import { BASE_MICROVERSE_URL } from './constants.js';
import createApp from './app.js';
export const getComments = async (movieId) => {
    const appId = await createApp();
    try {
        const response = await fetch(`${BASE_MICROVERSE_URL}/${appId}/comments?item_id=${movieId}`, {
            method: 'GET',
        });
        const data = await response.json();
        return data;
    } catch (error) {
        throw new Error('Could not fetch comments. Please try again later.');
    }

}


export const giveComments = async ({ movieId, username, comment }) => {
    const appId = await createApp();
    const response = await fetch(`${BASE_MICROVERSE_URL}/${appId}/comments`, {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify({
            item_id: movieId,
            username: username,
            comment: comment
        })
    });
    const data = await response.json();
    return data.result;
}