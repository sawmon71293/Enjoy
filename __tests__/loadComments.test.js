import { getComments } from '../src/modules/comments.js';
import counter from '../src/modules/helper.js';
import loadMovieComments from '../src/modules/loadComments.js';

jest.mock('../src/modules/comments.js');
jest.mock('../src/modules/helper.js');

describe('loadMovieComments', () => {
  beforeEach(() => {
    document.body.innerHTML = `<div
        id='movieModal'
        data-id=''
      ><div id='commentsSection'></div</div>`;
  });

  it('should load comments for the specified movie ID and update the modal', async () => {
    const movieId = 123;
    const mockComments = [
      {
        creation_date: '2022-01-01',
        username: 'user1',
        comment: 'Comment 1',
      },
      {
        creation_date: '2022-01-02',
        username: 'user2',
        comment: 'Comment 2',
      },
    ];
    getComments.mockResolvedValue(mockComments);
    counter.mockReturnValue(mockComments.length);
    const modal = document.querySelector('#movieModal');
    const commentsSection = modal.querySelector('#commentsSection');

    await loadMovieComments(movieId);

    expect(getComments).toHaveBeenCalledWith(movieId);

    expect(commentsSection.innerHTML).toMatchSnapshot();
  });

  it('should not update the modal if comments cannot be retrieved', async () => {
    const movieId = 456;
    getComments.mockResolvedValue(null);
    const modal = document.querySelector('#movieModal');
    const commentsSection = modal.querySelector('#commentsSection');

    await loadMovieComments(movieId);

    expect(getComments).toHaveBeenCalledWith(movieId);

    expect(commentsSection.innerHTML).toBe('');
  });
});
