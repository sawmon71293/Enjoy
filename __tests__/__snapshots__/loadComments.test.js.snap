// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadMovieComments should load comments for the specified movie ID and update the modal 1`] = `"<h6>Comments (2)</h6><p>2022-01-01 user1: Comment 1</p><p>2022-01-02 user2: Comment 2</p>"`;
